test/test_optim.py

```diff
@@ -1558,6 +1560,101 @@ def test_empty_grad(self):
                 self.assertEqual(original_param, param)
 
 
+    def test_post_hook(self):
```

Comment:
The test seems to fail only on the dynamo shard. You can add @skipIfTorchDynamo on these tests and open an issue with module: dynamo for the right people to look into this.